app/src/main/java/org/p2p/wallet/sell/ui/payload/SellPayloadPresenter.kt

```diff
@@ -74,6 +74,7 @@ class SellPayloadPresenter(
         launch {
             try {
                 view.showLoading(isVisible = true)
+                needCheckForSellLock = true
                 // call order is important!
                 checkForSellLock()
                 sellInteractor.getTokenForSell().also {
```

Comment:
that's so weird. 

Looks like you need to clean this method and make it more simple, it's overloaded rn

Reply:
we have a tech-debt task for refactoring.
yep, it's overloaded